### PR TITLE
Optimize (compress) ct_report logs on CI and read them using a service worker in Browser

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-2283 "Optimize (compress) ct_report logs somehow on CI"

Proposed changes include:
- Compress files into tar.gz.
- We use https://github.com/esl/html-zip-reader to read the archive and serve it using a service worker.
- The logic for the service worker could be reviewed in this PR: https://github.com/esl/html-zip-reader/pull/1

- We can maybe stop using s3-parallel-put (unless it is smaller than awscli tools). But in a separate PR :)

__DO not forget to activate the latest version of the service worker, if you tried it before (to @NelsonVides ):__

https://esl.github.io/html-zip-reader/main.html (click uninstall, install).
(If u see "oops, not found" errors).

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236503) / elasticsearch_and_cassandra_mnesia / 86f46bc9867b9b241882bf3467f9690a63dd5215
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/236503/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4367/236503/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4000bb27a6159a.2024-08-27_19.53.01/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236734) / elasticsearch_and_cassandra_mnesia / 2c25ffeb2b5d6f423e3351d2fad05b2a13f0caa7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/236734/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4367/236734/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4052b800fd04f0.2024-08-28_14.06.47/index.html)
**OK: 437** / **Failed: 2** / **User-skipped: 47** / Auto-skipped: 0 
<details><summary>gdpr_SUITE:retrieve_personal_data:retrieve_personal_data_pubsub:retrieve_all_pubsub_data</summary>


```erlang
{error,{#{what => invalid_response_code,expected_type => ok,
      response_code => {exit_status,1}},
    [{graphql_helper,assert_response_code,2,
             [{file,"/home/circleci/project/big_tests/tests/graphql_helper.erl"},
              {line,258}]},
     {graphql_helper,get_ok_value,2,
             [{file,"/home/circleci/project/big_tests/tests/graphql_helper.erl"},
              {line,241}]},
     {gdpr_SUITE,request_and_unzip_personal_data,3,
           [{file,"/home/circleci/project/big_tests/tests/gdpr_SUITE.erl"},
            {line,1692}]},
     {gdpr_SUITE,retrieve_and_validate_personal_data,5,
           [{file,"/home/circleci/project/big_tests/tests/gdpr_SUITE.erl"},
            {line,1575}]},
     {gdpr_SUITE,'-retrieve_all_pubsub_data/1-fun-1-',3,
           [{file,"/home/circleci/project/big_tests/tests/gdpr_SUITE.erl"},
            {line,1273}]},
     {escalus_story,story,4,
            [{file,"/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
             {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1794}]},
     {test_server,run_test_case_eval1,6,
            [{file,"test_server.erl"},{line,1303}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4367/236734/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4052b800fd04f0.2024-08-28_14.06.47/big_tests.tests.gdpr_SUITE.logs/run.2024-08-28_14.08.47/gdpr_suite.retrieve_all_pubsub_data.html)
</details>
<details><summary>gdpr_SUITE:retrieve_personal_data:retrieve_personal_data_private_xml:retrieve_private_xml</summary>


```erlang
{error,{#{what => invalid_response_code,expected_type => ok,
      response_code => {exit_status,1}},
    [{graphql_helper,assert_response_code,2,
             [{file,"/home/circleci/project/big_tests/tests/graphql_helper.erl"},
              {line,258}]},
     {graphql_helper,get_ok_value,2,
             [{file,"/home/circleci/project/big_tests/tests/graphql_helper.erl"},
              {line,241}]},
     {gdpr_SUITE,request_and_unzip_personal_data,3,
           [{file,"/home/circleci/project/big_tests/tests/gdpr_SUITE.erl"},
            {line,1692}]},
     {gdpr_SUITE,retrieve_and_validate_personal_data,5,
           [{file,"/home/circleci/project/big_tests/tests/gdpr_SUITE.erl"},
            {line,1575}]},
     {escalus_story,story,4,
            [{file,"/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
             {line,72}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1794}]},
     {test_server,run_test_case_eval1,6,
            [{file,"test_server.erl"},{line,1303}]},
     {test_server,run_test_case_eval,9,
            [{file,"test_server.erl"},{line,1235}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4367/236734/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4052b800fd04f0.2024-08-28_14.06.47/big_tests.tests.gdpr_SUITE.logs/run.2024-08-28_14.08.47/gdpr_suite.retrieve_private_xml.html)
</details>

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236960) / elasticsearch_and_cassandra_mnesia / 6d24e88dca9537308989c2af164614ce4d940fb5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/236960/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](/ct_run.test%40192e1a8c89c3.2024-09-05_08.29.12/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236989) / elasticsearch_and_cassandra_mnesia / 2c95a7f8336137adf02c5d499aed894a85157654
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/236989/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4367/236989/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big.tar.gz//ct_run.test%40ca80a142f1a9.2024-09-05_09.04.31/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/237026) / elasticsearch_and_cassandra_mnesia / 5e414a0553d5993708b7e2ad4d29bf0dce6ec0b3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/237026/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4367/237026/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big.tar.gz//ct_run.test%4080fe34622610.2024-09-05_09.15.10/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/237061) / elasticsearch_and_cassandra_mnesia / 694f1b830d2cdc2d65940e34bdb88e8089b0b38d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/237061/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4367/237061/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big.tar.gz//ct_run.test%4038fcaa32134d.2024-09-05_09.43.41/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/237121) / elasticsearch_and_cassandra_mnesia / 3e60a26ce46d25a1b042b0a5a0bce0c50e3703b1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/237121/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4367/237121/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big.tar.gz//ct_run.test%4054f2a947045a.2024-09-05_10.32.31/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

File: tools/helpers.sh [hunk 1]
```diff
@@ -31,3 +31,13 @@ direct_s3_url() {
   local CT_REPORTS=${1:-}
   echo "https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/${CT_REPORTS}"
 }
+
+zip_reader_big_url() {
```
Comment (NelsonVides) [inline tools/helpers.sh:35]: Same, not `zip_`

Comment (NelsonVides) [reply tools/helpers.sh:35]: Also, aren't these repeated? Could we reuse code? See `circleci-helpers.sh`.

File: tools/circleci-helpers.sh [hunk 2]
```diff
@@ -30,3 +30,12 @@ direct_s3_url() {
   echo "https://circleci-mim-results.s3.eu-central-1.amazonaws.com/${CT_REPORTS}"
 }
 
+zip_reader_big_url() {
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply tools/circleci-helpers.sh:33]: yea, though zip means compression.
gz is gunzip ;) 
but could rename...

Oh, repo name is zip too.

[the remark this answers is not in the document]
Comment (NelsonVides) [reply tools/circleci-helpers.sh:33]: Yeah, oh, anyway 🤷🏽 
Still, these functions are repeated below in `./tools/helpers.sh`.

[the remark this answers is not in the document]
Comment (arcusfelis) [reply tools/circleci-helpers.sh:33]: but tools/circleci-helpers.sh is a copy of tools/helpers.sh to begin with :)

[the remark this answers is not in the document]
Comment (NelsonVides) [reply tools/circleci-helpers.sh:33]: Then why don't we unify them? 😳

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/237221) / elasticsearch_and_cassandra_mnesia / cfaeacaf552093325b3601e97de241591a0cd20a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/237221/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](/ct_run.test%40d1d03f1bc95e.2024-09-06_13.44.01/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/237245) / elasticsearch_and_cassandra_mnesia / 3fea460fb8fecd39fc8006bb108a72688701186e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4367/237245/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4367/237245/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big.tar.gz//ct_run.test%4038c4308bb00c.2024-09-06_13.54.16/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Review by NelsonVides — APPROVED** [review]
LGTM 🔥🔥🔥

**Pull request merged**